ztp/gitops-subscriptions/argocd/Upgrade.md

```diff
@@ -1,43 +1,127 @@
+
+### Remove obsolete content
+If the upgrade process results in obsolete policies, for example by moving policies from one namespace to another, the steps in this section should be taken prior to upgrade to remove those policies in an automated way. These steps need to be done prior to stopping the existing GitOps ZTP application (below) so the existing application can remove the affected resources.
+
+1. Remove the affected PolicyGenTemplate(s) from GIT, commit and push to the remote repository.
+1. Wait for the changes to synchronize through the application and the affected policies to be removed from the hub cluster.
+
+Note that removing these policies from GIT, and as a result also removing them from the hub cluster, will not affect any configuration of the managed spoke clusters. Removing a policy from the hub does not delete from the spoke cluster the CRs managed by that policy.
+
+### Stop the existing GitOps ZTP applications
+Removing the existing Applications ensures that any changes to
+existing content in the GIT repository is not rolled out until the new
+version of the tooling is in place.
+
+Using the application files from the deployment directory. Note, if you used custom names for the applications you must update the names in these files first. These commands perform a non-cascaded delete which leaves in place all generated resources.
 ```
-generators:
-# list of all the siteConfig*.yaml exist in the repository
-- site1.yaml
-- site2.yaml
-- site3.yaml
-``` 
- Save and push changes to the repository. The ArgoCD application will detect the changes that have been done to the SiteConfig git repository   
- and re-sync the cluster installation CRS to the hub cluster. Note: the installed clusters will not be re provisioned.
- 
-**Upgrading the policyGenTemplate repository:**
-In the policyGenTemplate repository follow the steps below:
- 1. Delete the post-sync.yaml and pre-sync.yaml files.
- 1. If any PolicyGenTemplate*.yaml in the repository contains a namespace definition, remove it and define the Namespace in separate file similar to the new [example/ns.yaml](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/gitops-subscriptions/argocd/example/policygentemplates/ns.yaml).
- 1. Adjust all Namespace names to start with prefix  `ztp`.
- 1. Adjust the PolicyGenTemplate*.yaml Namespace accordingly. 
- 1. Update the kustomization.yaml file to have all PGTs in the `generators` section, and all Namespaces in the `resources` section. For example:
+    $ oc delete -f clusters-app.yaml
+    $ oc delete -f policies-app.yaml
+```
+
+### Install Topology Aware Lifecycle Operator
+If not already installed, the [Topology Aware Lifecycle Operator](https://github.com/openshift-kni/cluster-group-upgrades-operator#readme) must be installed on the hub cluster. Refer to the steps under [Preparation of ZTP GIT repository](#preparation-of-ztp-git-repository) to install this operator.
```

Comment:
Refer to the steps under [Preparation of Hub cluster for ZTP]

Reply:
Thanks. Done.